DIO_Driver/DIO_Driver/Mcal/DIO/DIO_retval.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * DIO_retval.h
+ *
+ * Created: 7/7/2021 5:14:40 PM
+ *  Author: Mohamed Wagdy
+ */ 
+
+
+#ifndef __DIO_RETVAL_H__
+#define __DIO_RETVAL_H__
+
+/*- INCLUDES -----------------------------------------------*/
+#include "../../Infrastructure/STD_types.h"
+
+/*- PRIMITIVE TYPES ----------------------------------------*/
+typedef uint8_t DIO_ERROR_state_t;
+
+/*- ERRORS DEFINITIONS--------------------------------------*/
+
+/**
+* @brief: This return value indicate that the function
+* ran successfully and it returns no errors.
+*/
+#define           E_DIO_SUCCESS              (DIO_ERROR_state_t)( 0 )
```

Comment:
-Error Handling should be like this formula in the image below

![5](https://user-images.githubusercontent.com/44152298/127526433-13f0d095-0f85-41d0-9d0f-ecb781b79199.PNG)

[Uploading Sprints_CodeGuideLine.pdf…]()